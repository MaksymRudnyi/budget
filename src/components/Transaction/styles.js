import styled from 'styled-components';

export const Wrapper = styled.div`
    background: ${({ value }) => value < 0 ? '#e2e1e1' : '#fff'};
    border: 1px solid #dbdbdb;
    border-radius: 3px;
    padding: 5px;
    margin-bottom: 10px;
    display: flex;
    align-items: center;
    justify-content: space-between;
`;
Wrapper.displayName = 'TransactionWrapper'

export const TransactionDate = styled.div`
    flex-grow: 1;
`;
TransactionDate.displayName = 'TransactionDate';

export const Value = styled.div`
    flex-grow: 1;
`;
Value.displayName = 'Value'

export const Comment = styled.div`
    flex-grow: 2;
`;
Comment.displayName = 'Comment'

export const Icon = styled.span`
    img {
        max-width: 20px;
        margin-right: 10px;
    }
`;
Icon.displayName = 'Icon';

