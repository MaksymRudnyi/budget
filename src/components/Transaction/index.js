import PropTypes from 'prop-types';

import { Wrapper, TransactionDate, Value, Comment } from './styles';

const Transaction = ({ transaction: {value, date, comment} }) => (
    <Wrapper value={value}>
        <TransactionDate>{date}</TransactionDate>
        <Value>{value.toFixed(2)}</Value>
        <Comment>{comment}</Comment>
    </Wrapper>
);

Transaction.propTypes = {
    transaction: PropTypes.shape({
        label: PropTypes.string,
        value: PropTypes.number
    })
}

Transaction.defaultProps = {
    transaction: {
        label: '',
        value: 0
    }
}

export default Transaction