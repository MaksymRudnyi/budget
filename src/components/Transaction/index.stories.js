import React from 'react';

import Transaction from '.';

export default {
  title: 'Example/Transation',
  component: Transaction,
};

const Template = (args) => <Transaction {...args} />;

export const Income = Template.bind({});
Income.args = {
    transaction: {
        comment: 'some income',
        value: 23,
        date: '13.02.2021'
    }
};