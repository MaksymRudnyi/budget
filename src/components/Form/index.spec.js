import React from 'react';
import {shallow} from 'enzyme';
import Transaction from '.';

describe('Transaction component', () => {
    it('should show transaction', () => {
        const props = {
            transaction: {
                value: 23,
                comment: 'test',
                date: '01.01.2020'
            }
        }
        const sut = shallow(<Transaction {...props}/>);

        expect(sut).toMatchSnapshot();
    })
})