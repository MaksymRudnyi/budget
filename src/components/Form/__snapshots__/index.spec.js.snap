// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should show transaction 1`] = `
<TransactionWrapper>
  <form
    onSubmit={[Function]}
  >
    <Row>
      <Input
        name="value"
        onChange={[Function]}
        placeholder="Сума"
        type="number"
        value=""
      />
      <Input
        name="date"
        onChange={[Function]}
        type="date"
        value="2021-02-13"
      />
    </Row>
    <Row>
      <Button>
        Зберегти
      </Button>
      <Comment
        name="comment"
        onChange={[Function]}
        value=""
      />
    </Row>
  </form>
</TransactionWrapper>
`;
